### PR TITLE
feat(repository): add object repository dynamic return type extension

### PR DESCRIPTION
**Pull request by mazsudo**

### CODE REVIEW

File: src/Type/ObjectRepositoryDynamicReturnTypeExtension.php [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+ * @see PHPStan\Type\Doctrine\ObjectRepositoryDynamicReturnTypeExtension for original implementation.
+ *
+ * @author Benoit Maziere <benoit.maziere@ekino.com>
+ */
+class ObjectRepositoryDynamicReturnTypeExtension implements DynamicMethodReturnTypeExtension, BrokerAwareExtension
+{
+    /**
+     * @var Broker
+     */
+	private $broker;
```
Comment (mremi) [inline src/Type/ObjectRepositoryDynamicReturnTypeExtension.php]: 😱 indentation 🙈

File: src/Type/ObjectRepositoryDynamicReturnTypeExtension.php [hunk 2]
```diff
@@ -0,0 +1,122 @@
+<?php
+
+declare(strict_types = 1);
```
Comment (mremi) [inline src/Type/ObjectRepositoryDynamicReturnTypeExtension.php]: no whitespace I think around `=`

File: src/Type/ObjectRepositoryDynamicReturnTypeExtension.php [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        return BaseManager::class;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function isMethodSupported(MethodReflection $methodReflection): bool
+    {
+        $methodName = $methodReflection->getName();
+        return strpos($methodName, 'findBy') === 0
```
Comment (mremi) [inline src/Type/ObjectRepositoryDynamicReturnTypeExtension.php:67]: missing blank line before

**Comment by mremi** [comment]
Please update the CHANGELOG.md

**Comment by mazsudo** [comment]
are we good on this one @mremi ?

**Pull request merged**